### ISSUE TITLE
About Training Data Augmentation

### ISSUE THREAD

**Issue by CHIJIN-B** [AhmedZgaren/Save#2]
Thank you @AhmedZgaren for your hard work!
At first, I directly used the training code provided in the readme to train the model and found that the training speed was extremely fast. However, when I used the validation code to verify the model's performance, I found that it was far from the indicators in the paper.

<img width="573" height="42" alt="Image" src="https://github.com/user-attachments/assets/964f41e5-6e3a-4878-a3e1-a3c645493f8a" />
👆MAE on val set

<img width="564" height="31" alt="Image" src="https://github.com/user-attachments/assets/d53b62e5-f79f-49a5-96cf-8401440e651c" />
👆MAE on test set

Later, I discovered that the paper mentioned the content related to training image enhancement, which greatly increased the training dataset. But this part was not mentioned in the training file. Could you please share the specific implementation details of image enhancement? 
This is very important to me!

**Comment by AhmedZgaren** [comment]
Hello @CHIJIN-B, 

Thank you for your interest in SAVE. I have added the script augmentation under the utils folder. The function _mosaic_ combines four individual images to create a mosaic image. You can write your script to read images from the dataset and generate mosaic images. The count of images in each mosaic should be equal to the highest number among the combined images. 

Additionally, please remember to create a CSV file for the newly created images. If you have any additional questions, please don't hesitate to ask.

**Comment by CHIJIN-B** [comment]
Thank you @AhmedZgaren  for your reply!

Based on the augment.py you provided, I wrote a script for generating a mosaic dataset and the corresponding CSV file. The generated images are as follows:
![Image](https://github.com/user-attachments/assets/5e3f2801-d7ff-46ca-baa0-45a8b69db576)
![Image](https://github.com/user-attachments/assets/f4a6442a-6203-45fb-b2a6-35e857469324)

Afterwards, I trained using the original training set of 3,659 images and the generated dataset of 25,000 images, totaling 28,659 images. I set the epoch to 200, and the training loss changes are as follows:
<img width="1608" height="771" alt="Image" src="https://github.com/user-attachments/assets/f66c1aed-84b3-4c9d-9cbd-345a9d42b655" />

<img width="339" height="25" alt="Image" src="https://github.com/user-attachments/assets/71f69d49-fabe-4e95-bf10-dfb0c638f8f0" />👆MAE on val set
<img width="342" height="29" alt="Image" src="https://github.com/user-attachments/assets/9ad0e271-8c99-4932-8243-0ef67a83b0f0" />👆MAE on test set

I found that this was not much different from my previous validation effect without using the mosaic dataset, and noticed that the training dataset you mentioned in the paper had a total of 28,546 images instead of 28,659 images. Could it be that the quality of the mosaic dataset I generated deviates from the one you used, which led to my poorer training results?

If possible, could you please provide the generated dataset used for training at that time? This also has great reference value for my subsequent method comparison!
